evaluator.py

```diff
@@ -0,0 +1,186 @@
+            raise Exception(f"get_threshold(): Failed to parse evaluator config")
+
+    def calculate_gbps(self, result: dict, test_type: TestType) -> Bitrate:
+        if test_type == TestType.IPERF_TCP.name:
+            return self.calculate_gbps_iperf_tcp(result)
+        elif test_type == TestType.IPERF_UDP.name:
+            return self.calculate_gbps_iperf_udp(result)
+        elif test_type == TestType.HTTP.name:
+            return self.calculate_gbps_http(result)
+        else:
+            logger.error(f"Error calculating bitrate, Test of type {test_type} is not supported")
+            raise Exception(f"calculate_gbps(): Invalid test_type {test_type} provided")
+
+    def dump_to_json(self) -> dict:
+        passing = []
+        failing = []
+        for result in self.results:
+            output = {"test_case_id": result.test_id,
+                      "test_type": result.test_type,
+                      "Bitrate_Gbps_rx": result.bitrate_gbps.rx,
+                      "Bitrate_Gbps_tx": result.bitrate_gbps.tx}
+            if result.success == True:
+                passing.append(output)
+            else:
+                failing.append(output)
+        return {"passing": passing, "failing": failing}
+
+    def calculate_gbps_iperf_tcp(self, result: dict) -> Bitrate:
+        # If an error occured, bitrate = 0
+        if "error" in result:
+            logger.error(f"An error occured during iperf test: {result['error']}")
+            return Bitrate(0,0)
+
+        try:
+            sum_sent = result["end"]["sum_sent"]
+            sum_received = result["end"]["sum_received"]
+        except KeyError as e:
+            logger.error(f"KeyError: {e}. Malformed results when parsing iperf tcp for sum_sent/received")
+            raise Exception(f"calculate_gbps_iperf_tcp(): failed to parse iperf test results")
+
+        bitrate_sent = sum_sent["bits_per_second"] / 1e9
+        bitrate_received = sum_received["bits_per_second"] / 1e9
+
+        return Bitrate(float(f"{bitrate_sent:.5g}"), float(f"{bitrate_received:.5g}"))
+
+
+    def calculate_gbps_iperf_udp(self, result: dict) -> Bitrate:
+        # If an error occured, bitrate = 0
+        if "error" in result:
+            logger.error(f"An error occured during iperf test: {result['error']}")
+            return Bitrate(0,0)
+
+        sum_data = result["end"]["sum"]
+
+        # UDP tests only have sender traffic
+        bitrate_sent = sum_data["bits_per_second"] / 1e9
+        return Bitrate(float(f"{bitrate_sent:.5g}"), float(f"{bitrate_sent:.5g}"))
+
+
+    def calculate_gbps_http(self, result: dict) -> Bitrate:
+        #TODO: Add http traffic testing
+        return -1
+
+    def evaluate_pass_fail_status(self) -> Status:
+        total_passing = 0
+        total_failing = 0
+        for result in self.results:
+            if result.success:
```

Comment:
I'm not sure I follow, could you elaborate? This line is checking an array of objects of type Result() rather than in the case of ```if "error" in result:``` a dict made from iperf json output. I will update the class / variables to make this more clear.

Reply:
I see the difference, that makes sense. Thanks!